### PR TITLE
Move rabbit into its own bundle

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: .github/create_opm_index.sh [hunk 1]
```diff
@@ -9,8 +9,8 @@ echo "${INDEX_IMAGE_TAG}"
 echo "${BUNDLE_IMAGE}"
 echo "${STORAGE_BUNDLE_IMAGE}"
 
-echo "opm index add --bundles "${REGISTRY}/${BUNDLE_IMAGE}:${GITHUB_SHA}","${REGISTRY}/${STORAGE_BUNDLE_IMAGE}:${GITHUB_SHA}" --tag "${REGISTRY}/${INDEX_IMAGE}:${GITHUB_SHA}" -u podman --pull-tool podman"
-opm index add --bundles "${REGISTRY}/${BUNDLE_IMAGE}:${GITHUB_SHA}","${REGISTRY}/${STORAGE_BUNDLE_IMAGE}:${GITHUB_SHA}" --tag "${REGISTRY}/${INDEX_IMAGE}:${GITHUB_SHA}" -u podman --pull-tool podman
+echo "opm index add --bundles "${REGISTRY}/${BUNDLE_IMAGE}:${GITHUB_SHA}","${REGISTRY}/${STORAGE_BUNDLE_IMAGE}:${GITHUB_SHA}","quay.io/openstack-k8s-operators/rabbitmq-cluster-operator-bundle@sha256:9ff91ad3c9ef1797b232fce2f9adf6ede5c3421163bff5b8a2a462c6a2b3a68b" --tag "${REGISTRY}/${INDEX_IMAGE}:${GITHUB_SHA}" -u podman --pull-tool podman"
```
Comment (fmount) [inline .github/create_opm_index.sh:12]: +1 in general, and given you put here the rabbit bundle w/ its sha I suspect we don't rebuild it often. Is this a short term fix (e.g., until we create the `messaging` bundle for osp) or are you leaving `cluster-operator` by sha on purpose?

Anyway approving as it's a short term fix that unblock people, we can follow up on that front

**Comment by fmount** [comment]
I think we're still hitting the size limit as per the build-deploy log

**Comment by abays** [comment]
> I think we're still hitting the size limit as per the build-deploy log

Hmm, I don't see the `"request is too large"` error in the logs: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/324/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1661526516418220032/artifacts/openstack-operator-build-deploy/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-6eac618c70508e7c135084b46d87f2a293ed5ea9d4ca616280f0663afb1a3e77/namespaces/openshift-operator-lifecycle-manager/pods/catalog-operator-595d4787cd-svzbs/catalog-operator/catalog-operator/logs/current.log

File: .github/create_opm_index.sh (hunk 1 above)
Comment (dprince) [reply .github/create_opm_index.sh:12]: this is a short term fix. But I do also want to discuss "what if we just let all the bundles be for individual operators"?

**Comment by abays** [comment]
> > I think we're still hitting the size limit as per the build-deploy log
> 
> Hmm, I don't see the `"request is too large"` error in the logs: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/324/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1661526516418220032/artifacts/openstack-operator-build-deploy/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-6eac618c70508e7c135084b46d87f2a293ed5ea9d4ca616280f0663afb1a3e77/namespaces/openshift-operator-lifecycle-manager/pods/catalog-operator-595d4787cd-svzbs/catalog-operator/catalog-operator/logs/current.log

I think this is the reason for the failure:

```
2023-05-25T00:42:55.478107114Z I0525 00:42:55.477915       1 event.go:282] Event(v1.ObjectReference{Kind:"Namespace", Namespace:"", Name:"openstack", UID:"b69c86a7-a43d-44ad-82bd-a4505d0acb10", APIVersion:"v1", ResourceVersion:"59027", FieldPath:""}): type: 'Warning' reason: 'ResolutionFailed' constraints not satisfiable: subscription openstack-operator requires openstack-operator-index/openstack/alpha/openstack-operator.v0.0.1, subscription openstack-operator exists, bundle openstack-operator.v0.0.1 requires an operator with package: cluster-operator and with version in range: >=0.0.0
```

I see that repeated a lot.  And I get the same thing when I build and run locally.

**Comment by dprince** [comment]
I've tested multiple times, now. The build from yesterday evening works fine in my environment. (previous build broken)

**Comment by abays** [comment]
> I've tested multiple times, now. The build from yesterday evening works fine in my environment. (previous build broken)

The package name for the dependency seems to be this in my environment:

```
$ oc get packagemanifest | grep rabbit
rabbitmq-cluster-operator                          Community Operators   14d
```

i.e. not just `cluster-operator`

File: .github/create_opm_index.sh (hunk 1 above)
Comment (fmount) [reply .github/create_opm_index.sh:12]: ack makes sense, let's merge this first and I'm +1 to follow up and put all the options on the table.

**Review by fmount — APPROVED** [review]

**Comment by fmount** [comment]
I guess when https://github.com/openshift/release/pull/39733 merges the job here will be green

**Comment by abays** [comment]
> I guess when [openshift/release#39733](https://github.com/openshift/release/pull/39733) merges the job here will be green

I had to change `dependencies.yaml` like so to get this to work when building locally.  Maybe it's just my environment:

```
diff --git a/dependencies.yaml b/dependencies.yaml
index 2f4f24e..f5b66b0 100644
--- a/dependencies.yaml
+++ b/dependencies.yaml
@@ -5,5 +5,5 @@ dependencies:
       version: ">0.0.0"
   - type: olm.package
     value:
-      packageName: cluster-operator
+      packageName: rabbitmq-cluster-operator
       version: ">=0.0.0"
```

**Comment by abays** [comment]
@dprince and I diagnosed my issue.  As a follow-up, we'll need to fix the `Makefile` to make local builds work.  You don't need to change the `dependencies.yaml`.

**Comment by abays** [comment]
https://github.com/openshift/release/pull/39733 has merged.  Retesting.

/test openstack-operator-build-deploy

**Pull request merged**